client/blocks/site/index.jsx

```diff
@@ -148,13 +178,32 @@ class Site extends React.Component {
 							<Gridicon icon="house" size={ 18 } />
 						</span>
 					) }
-				</a>
+				</button>
 				{ this.props.indicator ? <SiteIndicator site={ site } /> : null }
 			</div>
 		);
 	}
 }
 
-export default connect( ( state, { siteId, site } ) => ( {
-	site: siteId ? getSite( state, siteId ) : site,
-} ) )( localize( Site ) );
+function mapStateToProps( state ) {
+	const currentUser = getCurrentUser( state );
+	const selectedSiteId = getSelectedSiteId( state );
+	const isSingleSite = !! selectedSiteId || currentUser.site_count === 1;
+	const siteId = selectedSiteId || ( isSingleSite && getPrimarySiteId( state ) ) || null;
+	const site = getSite( state, siteId );
+
+	return {
+		isPreviewable: isSitePreviewable( state, siteId ),
+		siteId,
+		site,
+		siteSuffix: site ? '/' + site.slug : '',
```

Comment:
We usually call this prop `siteSlug`. And for `<Site>`, a valid site is always available, so there doesn't need to be any default for null `site`. That's needed only in sidebar, which handles the "all sites selected" case.